src/dmd/dtemplate.d

```diff
@@ -5650,15 +5656,13 @@ extern (C++) final class TemplateAliasParameter : TemplateParameter
     override RootObject defaultArg(Loc instLoc, Scope* sc)
     {
         RootObject da = defaultAlias;
-        Type ta = isType(defaultAlias);
-        if (ta)
-        {
-            if (ta.ty == Tinstance)
-            {
-                // If the default arg is a template, instantiate for each type
-                da = ta.syntaxCopy();
-            }
-        }
+
+        // Create a copy to not taint the default found in template
+        // declaration because it might depend on preceding parameters
+        if (Type ta = isType(defaultAlias))
+            da = ta.syntaxCopy();
+        else if (Expression ea = isExpression(defaultAlias))
+            da = ea.syntaxCopy();
```

Comment:
If this PR is stalled on more complicated matters, I think that this part is fairly obviously correct and fixes: https://issues.dlang.org/show_bug.cgi?id=3538 (which I had marked as a duplicate of 22540 since you were already trying to fix.)